model/buildlogger.go

```diff
@@ -109,9 +115,71 @@ func (l *Log) SaveNew() error {
 	return errors.Wrapf(err, "problem saving new log %s", l.ID)
 }
 
-// AppendLogChunkInfo adds a new log chunk to the log's chunks array in the
+// Remove removes the log from the database. The environment should not be nil.
+func (l *Log) Remove() error {
+	if l.env == nil {
+		return errors.New("cannot remove a log with a nil environment")
+	}
+	ctx, cancel := l.env.Context()
+	defer cancel()
+
+	if l.ID == "" {
+		l.ID = l.Info.ID()
+	}
+
+	_, err := l.env.GetDB().Collection(buildloggerCollection).DeleteOne(ctx, bson.M{"_id": l.ID})
```

Comment:
why is the first return value ignorable? I think it can fail to delete in some cases without returning errors.